apps/jobs/views.py

```diff
@@ -2301,26 +2302,152 @@ def get_submission_file_presigned_url(request, challenge_phase_pk):
             "request": request,
         },
     )
+    # Set default num of chunks to 1 if num of chunks is not specified
+    num_file_chunks = 1
+    if request.data.get("num_file_chunks"):
+        num_file_chunks = int(request.data["num_file_chunks"])
+
     if serializer.is_valid():
         serializer.save()
         submission = serializer.instance
 
         file_key_on_s3 = "{}/{}".format(
             settings.MEDIAFILES_LOCATION, submission.input_file.name
         )
-        response = generate_presigned_url(file_key_on_s3, challenge.pk)
+        response = generate_presigned_url_for_multipart_upload(file_key_on_s3, challenge.pk, num_file_chunks)
         if response.get("error"):
             response_data = response
             return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
         response_data = {
-            "presigned_url": response.get("presigned_url"),
+            "presigned_urls": response.get("presigned_urls"),
+            "upload_id": response.get("upload_id"),
             "submission_pk": submission.pk,
         }
         return Response(response_data, status=status.HTTP_201_CREATED)
     response_data = {"error": serializer.errors}
     return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
 
+@api_view(["POST"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def submission_file_upload_complete(request, challenge_phase_pk, submission_pk):
+    """
+    API to complete multipart upload of presigned url submission
+
+    Arguments:
+        request {HttpRequest} -- The request object
+        challenge_phase_pk {int} -- Challenge phase primary key
+        submission_pk {int} -- Submission primary key
+    Returns:
+         Response Object -- An object containing the presignd url and submission id, or an error message if some failure occurs
+    """
+    if settings.DEBUG or settings.TEST:
+        response_data = {"error": "Sorry, this feature is not available in development or test environment."}
+        return Response(response_data)
+
+    # Check if the challenge phase exists or not
+    try:
+        challenge_phase = get_challenge_phase_model(challenge_phase_pk)
```

Comment:
There is no need to use try/except with the get_challenge_phase_model function as it automatically handles that case. Please see its definition.

Reply:
@Ram81 Please resolve this as well.